src/components/recovery/CancelRecoveryButton/index.tsx

```diff
@@ -55,7 +55,7 @@ export function CancelRecoveryButton({
   return (
     <CheckWallet allowNonOwner>
       {(isOk) => {
-        const isDisabled = !isOk || (isGuardian && !isExpired)
+        const isDisabled = isOwner ? !isOk : !isOk && !isExpired
```

Comment:
This makes it possible to cancel a recovery tx as a non-owner before it has expired. I assume that tx would fail so I suggest to still disable it.

Reply:
I've tweaked the logic in 6c7fca83eb530143c7bee0be26e0ced1ff360022 to only disable if not expired as [non-owners are able to skip expired transactions](https://github.com/gnosis/zodiac-modifier-delay/blob/main/contracts/Delay.sol#L203)